### PR TITLE
apply scaling to input but not conditioning data 

### LINKED ISSUE #229: 🐛[BUG]: Resdiff conditional input scaling
### Version

n/a

### On which installation method(s) does this occur?

_No response_

### Describe the issue

I know resdiff isn't in the code base yet, but wanted to file this since I'll be gone.

Not a bug per-se, but @MortezaMardani, Arash, Jaideep and I have discussed an issue with how the conditional inputs are scaled in the EDMPreconditioner:
```
        #SR: concatenate input channels
        if img_lr is None:
            x = x
        else:
            x = torch.cat((x, img_lr), dim=1)

        x = x.to(torch.float32)
        sigma = sigma.to(torch.float32).reshape(-1, 1, 1, 1)
        class_labels = None if self.label_dim == 0 else torch.zeros([1, self.label_dim], device=x.device) if class_labels is None else class_labels.to(torch.float32).reshape(-1, self.label_dim)
        dtype = torch.float16 if (self.use_fp16 and not force_fp32 and x.device.type == 'cuda') else torch.float32

        c_skip = self.sigma_data ** 2 / (sigma ** 2 + self.sigma_data ** 2)
        c_skip = 0.0*c_skip
        c_out = sigma * self.sigma_data / (sigma ** 2 + self.sigma_data ** 2).sqrt()
        c_out = torch.ones_like(c_out)
        c_in = 1 / (self.sigma_data ** 2 + sigma ** 2).sqrt()
        c_in = torch.ones_like(c_in)
        c_noise = sigma.log() / 4
        c_noise = 0.0*c_noise
        
        F_x = self.model((c_in * x).to(dtype), c_noise.flatten(), class_labels=class_labels, **model_kwargs)
```

Specifically, for large noise c_in is very small. So both the conditional input "img_lr" and the latents are scaled down. I think we agreed the conditional inputs should probably not scaled down like this (i.e. the class labels are not).

The code would be modified like this:
```
        x = x.to(torch.float32)
        sigma = sigma.to(torch.float32).reshape(-1, 1, 1, 1)
        class_labels = (
            None
            if self.label_dim == 0
            else torch.zeros([1, self.label_dim], device=x.device)
            if class_labels is None
            else class_labels.to(torch.float32).reshape(-1, self.label_dim)
        )
        dtype = (
            torch.float16
            if (self.use_fp16 and not force_fp32 and x.device.type == "cuda")
            else torch.float32
        )

        c_skip = self.sigma_data**2 / (sigma**2 + self.sigma_data**2)
        c_out = sigma * self.sigma_data / (sigma**2 + self.sigma_data**2).sqrt()
        c_in = 1 / (self.sigma_data**2 + sigma**2).sqrt()
        c_noise = sigma.log() / 4

        arg = c_in * x

        if img_lr is not None:
            arg = torch.cat([arg, img_lr], dim=1)

        F_x = self.model(
            arg.to(dtype),
            c_noise.flatten(),
            class_labels=class_labels,
            **model_kwargs,
        )
```

cc @mnabian 

### Minimum reproducible example

_No response_

### Relevant log output

_No response_

### Environment details

_No response_

### PR DESCRIPTION
**Pull request by yairchn**
# Modulus Pull Request

## Description
This PR closes #229; where the scaling was originally applied to both the input `x` and the low resolution conditioning data `img_lr` and its now applied only to `x`. 

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/modulus/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [x] The [CHANGELOG.md](https://github.com/NVIDIA/modulus/blob/main/CHANGELOG.md) is up to date with these changes.
- [x] An [issue](https://github.com/NVIDIA/modulus/issues) is linked to this pull request.

### CODE REVIEW

**Review by MortezaMardani — APPROVED** [review]

File: modulus/models/diffusion/preconditioning.py [hunk 1]
```diff
@@ -780,6 +780,8 @@ def forward(
         c_in = 1 / (self.sigma_data**2 + sigma**2).sqrt()
         c_noise = sigma.log() / 4
 
+        x = torch.cat((x, img_lr), dim=1)
+
         F_x = self.model(
             (c_in * x).to(dtype),
```
Comment (nbren12) [inline modulus/models/diffusion/preconditioning.py:786]: This looks incorrect. Is the latent `x` being scaled by `c_in` twice?

Suggest using this implementation, which is in my EDM code base

```
        if condition is None:
            arg = c_in * x
        else:
            condition = condition.to(torch.float32)
            arg = torch.cat([c_in * x, condition], dim=1)

        F_x = self.model(
            arg.to(dtype), c_noise.flatten(), class_labels=class_labels, **model_kwargs
        )
        assert F_x.dtype == dtype
        D_x = c_skip * x + c_out * F_x.to(torch.float32)
        return D_x
```

**Review by nbren12 — COMMENTED** [review]
I don't think the PR is correct.

File: modulus/models/diffusion/preconditioning.py (hunk 1 above)
Comment (nbren12) [reply modulus/models/diffusion/preconditioning.py:786]: easier to follow IMO without overwriting the name "x". That is why I instead name it `arg`.
